marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryTransactionTest.java

```diff
@@ -190,4 +197,31 @@ public void testMultipleCommit()
         assertEquals("expected 4",conn.size(), 4L);
     }
 
+    // https://github.com/marklogic/marklogic-sesame/issues/358
+    @Test
+    public void testTransactionAccessForMultiModelData()
+    {
+        Resource context3 = conn.getValueFactory().createIRI("http://marklogic.com/test/context3");
+        ValueFactory f = conn.getValueFactory();
+        IRI alice = f.createIRI("http://example.org/people/alice");
+        IRI name = f.createIRI("http://example.org/ontology/name");
+        IRI person = f.createIRI("http://example.org/ontology/Person");
+        Literal alicesName = f.createLiteral("Alice");
+
+        conn.begin();
+        conn.add(alice, name, alicesName, context3);
+        conn.add(alice, RDF.TYPE, person, context3);
+
+        File file = new File("src/test/resources/testdata/bbq1.xml");
+        DocumentManager docMgr = adminClient.newDocumentManager();
+        String docId = "/tx-rollback/bbq1.xml";
+        FileHandle handle = new FileHandle(file);
+        handle.set(file);
+        handle.setFormat(Format.XML);
+        docMgr.write(docId, handle, conn.getTransaction());
+        DocumentDescriptor documentDescriptor = docMgr.exists(docId, conn.getTransaction());
+        Assert.assertEquals(docId, documentDescriptor.getUri());
+        Assert.assertNull(docMgr.exists(docId));
+        conn.rollback();
```

Comment:
I think this needs just two more assertions.  After the rollback, con.getTransaction() should be null (yes?) and the document must not exist, just like in line 224.  After that this test will be all good.